demos/multi_material/compositional_buoyancy/van_keken_1997.py

```diff
@@ -0,0 +1,276 @@
+# \right]$$
+#
+# This example
+# ------------
+#
+# Here, we consider the isoviscous Rayleigh-Taylor instability presented in van Keken
+# et al. (1997). Inside a 2-D domain, a buoyant, lighter material sits beneath a denser
+# material. The initial material interface promotes the development of a rising
+# instability on the domain's left-hand side, and further smaller-scale convective
+# dynamics take place throughout the remainder of the simulation.
+
+# As with all examples, the first step is to import the gadopt module, which
+# provides access to Firedrake and associated functionality.
+
+from gadopt import *
+
+# We next set up the mesh and function spaces and specify functions to hold our
+# solutions, as in our previous tutorials.
+
+# +
+nx, ny = 40, 40  # Number of cells in x and y directions
+lx, ly = 0.9142, 1  # Domain dimensions in x and y directions
+# Rectangle mesh generated via Firedrake
+mesh = RectangleMesh(nx, ny, lx, ly, quadrilateral=True)
+left_id, right_id, bottom_id, top_id = 1, 2, 3, 4  # Boundary IDs
+
+V = VectorFunctionSpace(mesh, "CG", 2)  # Velocity function space (vector)
+W = FunctionSpace(mesh, "CG", 1)  # Pressure function space (scalar)
+Z = MixedFunctionSpace([V, W])  # Stokes function space (mixed)
+Q = FunctionSpace(mesh, "CG", 2)  # Temperature function space (scalar)
+K = FunctionSpace(mesh, "DQ", 2)  # Level-set function space (scalar, discontinuous)
+R = FunctionSpace(mesh, "R", 0)  # Real space for time step
+
+z = Function(Z)  # A field over the mixed function space Z
+u, p = split(z)  # Symbolic UFL expressions for velocity and pressure
+z.subfunctions[0].rename("Velocity")  # Associated Firedrake velocity function
+z.subfunctions[1].rename("Pressure")  # Associated Firedrake pressure function
+T = Function(Q, name="Temperature")  # Firedrake function for temperature
+psi = Function(K, name="Level set")  # Firedrake function for level set
+# -
+
+# We now provide initial conditions for the level set field. To this end, we use the
+# shapely library to represent the initial location of the material interface and
+# derive the signed-distance function. Finally, we apply the transformation to obtain a
+# smooth step function profile.
+
+# +
+import numpy as np
+import shapely as sl
+
+
+def cosine_curve(x, amplitude, wavelength, vertical_shift):
+    """Cosine curve equation with an amplitude and a vertical shift"""
+    return amplitude * np.cos(2 * np.pi / wavelength * x) + vertical_shift
+
+
+interface_deflection = 0.02  # Amplitude of the cosine function marking the interface
+interface_wavelength = 2 * lx  # Wavelength of the cosine function
+material_interface_y = 0.2  # Vertical shift of the interface along the y axis
+# Group parameters defining the cosine profile
+isd_params = (interface_deflection, interface_wavelength, material_interface_y)
+
+# Shapely LineString representation of the material interface
+interface_x = np.linspace(0, lx, 1000)
```

Comment:
Out of interest, are results at all sensitivity to the choice of 1000 here? Obviously irrelevant for this tutorial, but perhaps when we think about publishing this approach. 

Probably a comment here too about why you chose 1000?

Reply:
This number defines the amount of points used to discretise the initial interface. You just need a number that is sufficiently large to capture the wiggles, if any, of the initial interface. If you put something too low, your initial state will not look like what you expected on ParaView.

Reply:
Does does the discretised interface ultimately smear to some extent, and this 1000 is just to ensure the best possible representation of that initial condition with our current discretisation?